lib/ecto/changeset.ex

```diff
@@ -2926,23 +3014,23 @@ defmodule Ecto.Changeset do
   end
 end
 
-defimpl Inspect, for: Ecto.Changeset do
-  import Inspect.Algebra
-
-  def inspect(changeset, opts) do
-    list = for attr <- [:action, :changes, :errors, :data, :valid?] do
-      {attr, Map.get(changeset, attr)}
-    end
-
-    container_doc("#Ecto.Changeset<", list, ">", opts, fn
-      {:action, action}, opts   -> concat("action: ", to_doc(action, opts))
-      {:changes, changes}, opts -> concat("changes: ", to_doc(changes, opts))
-      {:data, data}, _opts      -> concat("data: ", to_struct(data, opts))
-      {:errors, errors}, opts   -> concat("errors: ", to_doc(errors, opts))
-      {:valid?, valid?}, opts   -> concat("valid?: ", to_doc(valid?, opts))
-    end)
-  end
-
-  defp to_struct(%{__struct__: struct}, _opts), do: "#" <> Kernel.inspect(struct) <> "<>"
-  defp to_struct(other, opts), do: to_doc(other, opts)
-end
+# defimpl Inspect, for: Ecto.Changeset do
+#   import Inspect.Algebra
+
+#   def inspect(changeset, opts) do
+#     list = for attr <- [:action, :changes, :errors, :data, :valid?] do
+#       {attr, Map.get(changeset, attr)}
+#     end
+
+#     container_doc("#Ecto.Changeset<", list, ">", opts, fn
+#       {:action, action}, opts   -> concat("action: ", to_doc(action, opts))
+#       {:changes, changes}, opts -> concat("changes: ", to_doc(changes, opts))
+#       {:data, data}, _opts      -> concat("data: ", to_struct(data, opts))
+#       {:errors, errors}, opts   -> concat("errors: ", to_doc(errors, opts))
+#       {:valid?, valid?}, opts   -> concat("valid?: ", to_doc(valid?, opts))
+#     end)
+#   end
+
+#   defp to_struct(%{__struct__: struct}, _opts), do: "#" <> Kernel.inspect(struct) <> "<>"
+#   defp to_struct(other, opts), do: to_doc(other, opts)
+# end
```

Comment:
It will be before merging. The inspect protocol override hides valuable data in the changeset which makes it harder to debug issues.

Reply:
@TheFirstAvenger Wait, who said that `inspect` protocol needs to be good for debugging? For this just write code like:

```elixir
changeset |> Map.from_struct() |> IO.inspect(label: "All map keys and values from Ecto.Changeset struct")
```

Reply:
How I work through this, I add and remove dozens of IO.inspects an hour, not always knowing which variable might be a changeset and which might not (in many cases it could be either). It is helpful for me during development to have this commented out, but as mentioned I will remove it when we are closer to being ready to merge.